ext/url/00_url.js

```diff
@@ -65,6 +65,20 @@ function opUrlParse(href, maybeBase) {
   return getSerialization(status, href, maybeBase);
 }
 
+function opUrlCanParse(href, maybeBase) {
+  let status;
+  if (maybeBase === undefined) {
+    status = ops.op_url_parse(href, componentsBuf);
+  } else {
+    status = ops.op_url_parse_with_base(
+      href,
+      maybeBase,
+      componentsBuf,
+    );
+  }
+  return status === 0 || status === 1;
+}
+
```

Comment:
Nitpick: this doesnt really seem necessary to be a separate function

Reply:
I want to prevent to call `getSerialization` in `URL.canParse`, should I put a flag on the `opUrlParse` argument?

Reply:
Let me think about this for a moment.